src/I18nProvider.js

```diff
@@ -0,0 +1,68 @@
+import React, { PureComponent } from 'react';
+import { createIntl, createIntlCache, RawIntlProvider, FormattedMessage } from 'react-intl';
+
+import { getTranslationPath } from './utils';
+import NotInitialisedError from './NotInitialisedError';
+
+export const FALLBACK_LANGUAGE = 'en';
+
+let translate = NotInitialisedError;
```

Comment:
Not sure if this is a clean solution, using this approach, your translation function is not correctly scoped to the provider it's in? I think theoretically you can now use the translate function outside of the provider

Reply:
To be honest, I do not like the idea either. But this is the way it is implemented right now as well and I don't have any clue if there is a better way to do so.

The reason why we do this, is to be able to export the `translate` method and use it outside of the provider scope, such as in sagas. So what we do here is exporting a variable instead of a const that changes during the mounting of the provider.

Reply:
I see, not a fan of this approach at all, but I understand its purpose now

Reply:
Feel free to propose a better approach :P